### DISCUSSION TITLE
What is the significance of Extraction Length?

### DISCUSSION THREAD

**Discussion by w0mbatian** [mcguirepr89/BirdNET-Pi#552] [Q&A] [answered]
Hi. I am completely new to this so there are a few bits I don't understand.
Particularly extraction length and how it relates to recording length. 
I am guessing it might be related to the available computing power but I am not sure.
The extraction length box is empty by default. Is it then the same as recording length?
I am running BirdNET-Pi on a Pi 4B 2GB which is doing nothing else.
What would be recommended settings for recording length and extraction length?
I could not find any explanation of the advanced settings.
Ian

**Comment by ehpersonal38** [comment, marked as answer]
Hi there, welcome to the BirdNET-Pi community! 

There's a decent writeup of what all the configurable variables do which you can find here: https://github.com/mcguirepr89/BirdNET-Pi/blob/main/birdnet.conf-defaults

> What would be recommended settings for recording length and extraction length?

Generally, the slower your processor, the longer you want your recording length to be. This is because you always want the time it takes your system to analyze a clip to be less than the length of that clip. Otherwise your analysis just won't be able to keep up with the incoming recordings and will fall behind.

So 15 seconds (the default) is a good value that works for most people (it takes my Pi about 2 seconds to analyze a clip of that length). I think on the really cheap $5 Pi, Patrick (the creator of BirdNET-Pi) recommends people to change the length to 30+ seconds.  

For extraction length, it depends on how much context you want to hear when you playback a detected bird sound. 3 seconds will give you little context, so you will really only hear the main sound, where a longer length like 9 seconds will give you plenty of audio to listen to.
